### PR TITLE
Oauth login

### PR DESCRIPTION
**Pull request by supradeep2819**

### CODE REVIEW

File: core/social_auth.py [hunk 1]
```diff
@@ -61,10 +64,73 @@ def login(request, payload: SocialAuthLoginSchema):
         oauth.save()
     token, _ = Token.objects.get_or_create(user=user)
     user_id = user.id
-    queryset = User.objects.prefetch_related("organizations").get(id=user_id)
-    logger.debug(queryset)
-    response = {
-        "auth_token" :token.key,
+    #HACK: Hardcoded everything as of now need to figure out a way to work this
```
Comment (gane5hvarma) [inline core/social_auth.py:67]: what is the hack?

File: core/social_auth.py [hunk 2]
```diff
@@ -61,10 +64,73 @@ def login(request, payload: SocialAuthLoginSchema):
...
-        "auth_token" :token.key,
+    #HACK: Hardcoded everything as of now need to figure out a way to work this 
+    result = urlparse(os.environ["DATABASE_URL"])
+    username = result.username
+    password = result.password
+    database = result.path[1:]
+    hostname = result.hostname
+    port = result.port
+    conn = psycopg2.connect(user=username, password=password, host=hostname, port=port,database=database)
+    query = f'SELECT * FROM core_user_organizations WHERE user_id = {user_id}'
```
Comment (gane5hvarma) [inline core/social_auth.py:75]: What are we doing with this query. Can you remove this line if its not required

File: core/social_auth.py [hunk 3]
```diff
@@ -61,10 +64,73 @@ def login(request, payload: SocialAuthLoginSchema):
...
+    """
+    cursor.execute(query,(user_id,))
+    result = cursor.fetchone()
+    cursor.close()
+    conn.close()
+    response = {    
+        "auth_token": token.key,
+        "organizations":result
     }
+    logger.debug("result is ---------------------------------------")
```
Comment (gane5hvarma) [inline core/social_auth.py:132]: Dont think this is needed, can you remove this line

File: core/social_auth.py [hunk 4]
```diff
@@ -26,7 +29,7 @@ def generate_key():
 
 
 # TODO response for bad request, 400
-@router.post("/login", response={200: AuthToken})
+@router.post("/login", response={200: Json})
```
Comment (gane5hvarma) [inline core/social_auth.py:32]: can you handle bad request as well

**Review by gane5hvarma — CHANGES_REQUESTED** [review]

File: core/social_auth.py (hunk 4 above)
Comment (supradeep2819) [reply core/social_auth.py]: resolved

File: core/social_auth.py (hunk 3 above)
Comment (supradeep2819) [reply core/social_auth.py]: resolved

File: core/social_auth.py (hunk 1 above)
Comment (supradeep2819) [reply core/social_auth.py]: resolved

File: core/social_auth.py (hunk 2 above)
Comment (supradeep2819) [reply core/social_auth.py]: resolved

**Pull request closed without merging**